tools/schemacode/bidsschematools/render.py

```diff
@@ -287,9 +287,32 @@ def make_filename_template(schema, n_dupes_to_combine=6, **kwargs):
     paragraph = paragraph.rstrip()
     codeblock = "Template:\n```Text\n" + paragraph + "\n```"
     codeblock = codeblock.expandtabs(4)
+    codeblock = append_filename_template_legend(codeblock)
     return codeblock
 
 
+def append_filename_template_legend(text=None):
+
+    if text is None:
+        return
+
+    text += """
+
+<details>
+<summary><strong>Legend:</strong></summary>
+
+- Filename entities or folders between square brackets (for example `[_ses-<label>]`) are OPTIONAL.
+- See the glossary for a definition of [`label`](../99-appendices/14-glossary.md#label)
+  and [`index`](../99-appendices/14-glossary.md#index).
+- See the [entity page](../99-appendices/09-entities.md) for a definition of each entity.
```

Comment:
In combination with #1228, I think you can drop the second and third bullet points. I also think it would be good to have the following:

- Some entities may only allow specific values, in which case those values are listed in `<>`, separated by `|`.
- `_<suffix>` means that there are several (>6) valid suffixes for this filename pattern.
- `.<extension>` means that there are several (>6) valid extensions for this file type.
- `[.gz]` means that both the unzipped and gzipped version of the extension are valid.

Reply:
The additions you suggest make sense. Will wait for those other PRs to be merged then I will add those.

Reply:
done